patches/minecraft/net/minecraft/client/gui/advancements/GuiScreenAdvancements.java.patch

```diff
@@ -0,0 +1,100 @@
+     private int field_191941_t;
+     private int field_191942_u;
+     private boolean field_191944_v;
++    private static int tabPage;
++    private static int maxPages;
+ 
+     public GuiScreenAdvancements(ClientAdvancementManager p_i47383_1_)
+     {
+@@ -49,6 +52,14 @@
+         {
+             this.field_191946_h.func_194230_a(this.field_191940_s == null ? null : this.field_191940_s.func_193935_c(), true);
+         }
++        if (this.field_191947_i.size() > AdvancementTabType.field_192659_e)
++        {
++            int guiLeft = (this.field_146294_l - 252) / 2;
++            int guiTop = (this.field_146295_m - 140) / 2;
++            this.field_146292_n.add(new GuiButton(101, guiLeft, guiTop - 50, 20, 20, "<"));
++            this.field_146292_n.add(new GuiButton(102, guiLeft + 252 - 20, guiTop - 50, 20, 20, ">"));
++            maxPages = this.field_191947_i.size() / AdvancementTabType.field_192659_e;
++        }
+     }
+ 
+     public void func_146281_b()
+@@ -71,7 +82,7 @@
+ 
+             for (GuiAdvancementTab guiadvancementtab : this.field_191947_i.values())
+             {
+-                if (guiadvancementtab.func_191793_c(i, j, p_73864_1_, p_73864_2_))
++                if (guiadvancementtab.getPage() == tabPage && guiadvancementtab.func_191793_c(i, j, p_73864_1_, p_73864_2_))
+                 {
+                     this.field_191946_h.func_194230_a(guiadvancementtab.func_193935_c(), true);
+                     break;
+@@ -122,6 +133,14 @@
+         this.func_146276_q_();
+         this.func_191936_c(p_73863_1_, p_73863_2_, i, j);
+         this.func_191934_b(i, j);
++        super.func_73863_a(p_73863_1_, p_73863_2_, p_73863_3_);
++        if (maxPages != 0)
++        {
++            String page = String.format("%d / %d", tabPage + 1, maxPages + 1);
++            int width = this.field_146289_q.func_78256_a(page);
++            GlStateManager.func_179140_f();
++            this.field_146289_q.func_175063_a(page, i + (252 / 2) - (width / 2), j - 44, -1);
++        }
+         this.func_191937_d(p_73863_1_, p_73863_2_, i, j);
+     }
+ 
+@@ -163,7 +182,8 @@
+ 
+             for (GuiAdvancementTab guiadvancementtab : this.field_191947_i.values())
+             {
+-                guiadvancementtab.func_191798_a(p_191934_1_, p_191934_2_, guiadvancementtab == this.field_191940_s);
++                if(guiadvancementtab.getPage() == tabPage)
++                    guiadvancementtab.func_191798_a(p_191934_1_, p_191934_2_, guiadvancementtab == this.field_191940_s);
+             }
+ 
+             GlStateManager.func_179091_B();
+@@ -172,7 +192,8 @@
+ 
+             for (GuiAdvancementTab guiadvancementtab1 : this.field_191947_i.values())
+             {
+-                guiadvancementtab1.func_191796_a(p_191934_1_, p_191934_2_, this.field_146296_j);
++                if(guiadvancementtab1.getPage() == tabPage)
++                    guiadvancementtab1.func_191796_a(p_191934_1_, p_191934_2_, this.field_146296_j);
```

Comment:
If you don't change the indentation, the patch will only be a one line addition instead of a deletion and addition.